apis/core/validation/blueprint.go

```diff
@@ -18,6 +22,47 @@ import (
 
 var landscaperScheme = runtime.NewScheme()
 
+var importTypesWithExpectedConfig = map[string][]string{
+	string(core.ImportTypeData):   {"Schema"},
+	string(core.ImportTypeTarget): {"TargetType"},
+}
+var exportTypesWithExpectedConfig = map[string][]string{
+	string(core.ExportTypeData):   {"Schema"},
+	string(core.ExportTypeTarget): {"TargetType"},
+}
+
+// isFieldValueDefinition lists the fields which are not directly part of an Import/ExportDefinition, but of the FieldValueDefinition
+// (usually the fields which are present in both import and export definitions)
+// This is required for the reflection used below
+// a map is used for easier 'contains' queries, the values are ignored
+var isFieldValueDefinition = map[string]bool{
+	"Schema":     true,
+	"TargetType": true,
+}
+
+func keys(m map[string][]string) []string {
+	res := make([]string, len(m))
+	for k := range m {
+		res = append(res, string(k))
+	}
+	return res
+}
+
+func relevantConfigFields() map[string]bool {
```

Comment:
why is this function needed?
It returns always the same value?

Reply:
Yes, it returns the union of the values of `importTypesWithExpectedConfig` and `exportTypesWithExpectedConfig`. We need this union to find out which config fields exist and thus need to be checked. For example: if the `type` is `target`, we expect `TargetType` to be set while `Schema` should not be set. We cannot simply verify that all other fields are not set, because some need to be set in any case, such as `Name` and `Type`.
Instead of using a positive list here, we could also turn it around and exclude `Name` and `Type` and then assume that all other fields are config and should not be set unless they are listed in the current value of `importTypesWithExpectedConfig`.

Do you get what I mean? WDYT?

Reply:
Turning it the other way around will lead to the check function becoming uglier though, because then we will have to iterate over all fields of the import/export definition instead of iterating over a fixed set of fields.

Reply:
hm ok.
maybe it makes sense to expose that as a variable and only calculate that once during startup in the`init`func.
And can you add a docstring to describe the generates map

Reply:
I just added a docstring, will have a look at computing the list only once during init.